questiontype.php

```diff
@@ -198,12 +217,27 @@ public function export_to_xml($question, qformat_xml $format, $extra = null): st
         $expout = parent::export_to_xml($question, $format, $extra);
         $expout .= '    <numrows>' . $format->xml_escape($question->options->numrows) . "</numrows>\n";
         $expout .= '    <numcolumns>' . $format->xml_escape($question->options->numcolumns) . "</numcolumns>\n";
-        $words = $question->options->words;
-        foreach ($words as $word => $value) {
+        $fs = get_file_storage();
+        foreach ($question->options->words as $word => $value) {
             $expout .= "    <word>\n";
             foreach (self::WORD_FIELDS as $xmlfield) {
-                $exportedvalue = $format->xml_escape($value->{$xmlfield});
-                $expout .= "        <$xmlfield>{$exportedvalue}</$xmlfield>\n";
+                if ($xmlfield === 'clue' || $xmlfield === 'feedback') {
+                    if (!isset($value->{$xmlfield})) {
+                        $value->{$xmlfield} = '';
+                    }
+                    $formatfield = $xmlfield . 'format';
+                    if (!isset($value->{$formatfield})) {
+                        $value->{$formatfield} = FORMAT_HTML;
+                    }
+                    $files = $fs->get_area_files($question->contextid, 'question', $xmlfield, $value->id);
+                    $expout .= "        <{$xmlfield} {$format->format($value->{$formatfield})}>\n";
+                    $expout .= '        ' . $format->writetext($value->{$xmlfield});
+                    $expout .= $format->write_files($files);
+                    $expout .= "        </{$xmlfield}>\n";
```

Comment:
I am looking at this and thinking "Why is there not a helper function $format->write_text_with_files()) with a nice API? like import_text_with_files, but there is not, so this code is right.